File: 백준/Silver/2606. 바이러스/바이러스.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const n = +input.shift();
input.shift();

const computer = Array.from({ length: n + 1 }, () => []);
const visited = new Array(n + 1).fill(0);

for (const item of input) {
  const [a, b] = item.split(" ").map(Number);
  computer[a].push(b);
  computer[b].push(a);
}

function bfs(start) {
  const queue = [];
  queue.push(start);
  visited[start] = 1;
  let cnt = 0;

  while (queue.length > 0) {
    const node = queue.shift();

    for (const i of computer[node]) {
      if (!visited[i]) {
        queue.push(i);
        visited[i] = 1;
        cnt++;
      }
    }
  }

  return cnt;
}

console.log(bfs(1));

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
def input():
  return sys.stdin.readline().rstrip()

c = int(input())
t = int(input())
graph = [[] for i in range(c+1)]
for _ in range(t):
  a, b = map(int, input().split())
  graph[a].append(b)
  graph[b].append(a)

visited = [False] * (c+1)
cnt = 0

def dfs(graph, v, visited):
  global cnt
  visited[v] = True
  for i in graph[v]:
    if not visited[i]:
      cnt += 1
      dfs(graph, i, visited)

dfs(graph, 1, visited)
print(cnt)